### PR TITLE
SONARJAVA-2739 Exclude simple cases of loggers with resource bundles

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-checks/src/test/files/checks/PrintfMisuseCheck.java [hunk 1]
```diff
@@ -177,6 +178,21 @@ void foo(Calendar c){
     slf4jLog.warn("message ", 1); // Noncompliant {{String contains no format specifiers.}}
     slf4jLog.warn("message {}", 1);
     slf4jLog.warn("message");
+
+    java.util.logging.Logger logger2 = java.util.logging.Logger.getLogger("som.foo", "som.foo.resources.i18n.LogMessages");
```
Comment (andrea-guarino-sonarsource) [inline java-checks/src/test/files/checks/PrintfMisuseCheck.java]: If I'm not wrong, this case would still raise an FP:
```
java.util.logging.Logger logger2;
logger2 = java.util.logging.Logger.getLogger("som.foo", "som.foo.resources.i18n.LogMessages");
logger2.log(java.util.logging.Level.WARNING, "som.foo.errorcode", 404);
```

File: java-checks/src/main/java/org/sonar/java/checks/PrintfMisuseCheck.java [hunk 2]
```diff
@@ -67,6 +76,30 @@ protected void onMethodInvocationFound(MethodInvocationTree mit) {
...
+      return false;
+    }
+    if (id.is(Tree.Kind.MEMBER_SELECT)) {
+      id = ((MemberSelectExpressionTree) id).identifier();
+    }
+    if (id.is(Tree.Kind.IDENTIFIER)) {
+      Tree decl = ((IdentifierTree) id).symbol().declaration();
+      if (decl != null && decl.is(Tree.Kind.VARIABLE)) {
+        VariableTree var = ((VariableTree) decl);
+        ExpressionTree init = var.initializer();
```
Comment (andrea-guarino-sonarsource) [inline java-checks/src/main/java/org/sonar/java/checks/PrintfMisuseCheck.java]: Should we check for reassignments as well? See my comments on tests below.

File: java-checks/src/test/files/checks/PrintfMisuseCheck.java (hunk 1 above)
Comment (benzonico) [reply java-checks/src/test/files/checks/PrintfMisuseCheck.java]: You are right but I would not care too much. We are talking about loggers here, reassigning them seems to me a corner case.

Comment (andrea-guarino-sonarsource) [reply java-checks/src/test/files/checks/PrintfMisuseCheck.java]: Agreed, LGTM then.
